client/my-sites/themes/themes-magic-search-card/index.jsx

```diff
@@ -192,6 +189,24 @@ class ThemesMagicSearchCard extends React.Component {
 		this.updateInput( updatedInput );
 	}
 
+	focusOnInput = () => {
+		this.refs[ 'url-search' ].focus();
+	}
+
+	clearSearch = () => {
+		this.updateInput( '' );
+		this.focusOnInput();
+	}
+
+	handleClickOutside() {
+		this.setState( { searchIsOpen: false } );
+	}
+
+	// TODO remember cursor position and put it back after focus
```

Comment:
> // TODO remember cursor position and put it back after focus

Not sure we would ever get around to this. And what about if user clicks on specific position on purpose, only for the cursor to jump to remembered position?

Maybe remove this comment.